code/game/atoms.dm

```diff
@@ -293,6 +301,69 @@ its easier to just keep the beam vertical.
 /atom/proc/fire_act()
 	return
 
+/atom/proc/scp914_act(var/mode) //DON'T OVERRIDE, override scp914_act_on_type instead
+	var/list/chosen_mode = modes_914[mode]
+	if(chosen_mode && chosen_mode.len > 0)
+		var/chosen = pick(chosen_mode)
+		var/atom/chosen_obj = new chosen(loc)
+		if(chosen_obj)
+			return scp914_act_on_type(chosen_obj)
+	else
+		switch(mode)
+			if(1) //TODO: 1 and 2 should be refactored to fit MATERIAL datums once they are ready
+				var/atom/choice = new parent_type
+				if(choice)
+					var/atom/second_choice = choice.parent_type
+					if(second_choice)
+						var/atom/atombychoice = new second_choice(loc)
+						qdel(choice)
+						if(atombychoice)
+							return scp914_act_on_type(atombychoice)
+			if(2)
+				var/choice = parent_type
+				if(choice)
+					var/atom/atombychoice = new choice(loc)
+					if(atombychoice)
+						return scp914_act_on_type(atombychoice)
+			if(3)
+				var/list/choices = subtypesof(parent_type)
+				if(choices && choices.len > 0)
+					for(var/atom/typo in choices)
+						var/atom/choico = new typo
+						if("[choico.parent_type]" != "[parent_type]")
+							choices -= typo
+						qdel(choico)
+					if(choices.len > 0)
+						var/choice = pick(choices)
+						if(choice)
+							var/atom/atombychoice = new choice(loc)
+							if(atombychoice)
+								return scp914_act_on_type(atombychoice)
+			if(4)
+				var/list/choices = subtypesof(src)
+				if(choices && choices.len > 0)
+					for(var/atom/typo in choices)
+						var/atom/choico = new typo
+						if("[choico.parent_type]" != "[type]")
+							choices -= typo
+						qdel(choico)
+					if(choices.len > 0)
+						var/choice = pick(choices)
+						if(choice)
+							var/atom/atombychoice = new choice(loc)
+							if(atombychoice)
+								return scp914_act_on_type(atombychoice)
+			if(5) //TODO: Very Fine should give one SCP component to an output object
+				var/choice = pick(subtypesof(src))
+				if(choice)
+					var/atom/atombychoice = new choice(loc)
+					if(atombychoice)
+						return scp914_act_on_type(atombychoice)
+	return
+
+/atom/proc/scp914_act_on_type(var/atom/output) //to get around various type requirements, input/intake is src so don't worry
```

Comment:
You might want to provide the mode used to select this type *and* have another API to programmatically select the type based on mode.

Reply:
oh right mode is important here